From 9bdd60ae340aca6d0be8f183fbce7b8a4674227a Mon Sep 17 00:00:00 2001
From: Alexandru Croitor <alexandru.croitor@qt.io>
Date: Wed, 10 Sep 2025 10:48:40 +0200
Subject: [PATCH 121/553] CMake: Fix double slash in SBOM install path

Use _qt_internal_path_join to avoid concatenating paths with empty
components.

Pick-to: 6.9 6.8
Task-number: QTBUG-134894
Change-Id: I62fe286dc56172af9a4293158a93e1699ad239ab
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
(cherry picked from commit 6c31cb31810796c1f38ce953a9f2df3c5755ff71)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 cmake/QtPublicSbomGenerationHelpers.cmake | 10 ++++++----
 cmake/QtPublicSbomHelpers.cmake           |  8 ++++----
 2 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/cmake/QtPublicSbomGenerationHelpers.cmake b/cmake/QtPublicSbomGenerationHelpers.cmake
index 64126f59a02..1e8f18c29f9 100644
--- a/cmake/QtPublicSbomGenerationHelpers.cmake
+++ b/cmake/QtPublicSbomGenerationHelpers.cmake
@@ -85,10 +85,12 @@ function(_qt_internal_sbom_begin_project_generate)
 
     _qt_internal_sbom_get_git_version_vars()
 
-    set(default_sbom_file_name
-        "${arg_PROJECT}/${arg_PROJECT}-sbom-${QT_SBOM_GIT_VERSION_PATH}.spdx")
-    set(default_install_sbom_path
-        "\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/${default_sbom_file_name}")
+    _qt_internal_path_join(default_sbom_file_name
+        "${arg_PROJECT}" "${arg_PROJECT}-sbom-${QT_SBOM_GIT_VERSION_PATH}.spdx")
+
+    _qt_internal_path_join(default_install_sbom_path
+        "\${CMAKE_INSTALL_PREFIX}/" "${CMAKE_INSTALL_DATAROOTDIR}" "${default_sbom_file_name}"
+    )
 
     _qt_internal_sbom_set_default_option_value(OUTPUT "${default_install_sbom_path}")
     _qt_internal_sbom_set_default_option_value(OUTPUT_RELATIVE_PATH
diff --git a/cmake/QtPublicSbomHelpers.cmake b/cmake/QtPublicSbomHelpers.cmake
index 3935b917609..17f313e1cc9 100644
--- a/cmake/QtPublicSbomHelpers.cmake
+++ b/cmake/QtPublicSbomHelpers.cmake
@@ -168,12 +168,12 @@ function(_qt_internal_sbom_begin_project)
         ${compute_project_file_name_args}
     )
 
-    set(repo_spdx_relative_install_path
-        "${arg_INSTALL_SBOM_DIR}/${repo_project_file_name}")
+    _qt_internal_path_join(repo_spdx_relative_install_path
+        "${arg_INSTALL_SBOM_DIR}" "${repo_project_file_name}")
 
     # Prepend DESTDIR, to allow relocating installed sbom. Needed for CI.
-    set(repo_spdx_install_path
-        "\$ENV{DESTDIR}${install_prefix}/${repo_spdx_relative_install_path}")
+    _qt_internal_path_join(repo_spdx_install_path
+        "\$ENV{DESTDIR}${install_prefix}" "${repo_spdx_relative_install_path}")
 
     if(arg_LICENSE_EXPRESSION)
         set(repo_license "${arg_LICENSE_EXPRESSION}")
-- 
2.51.2

